testdata/p4_16_samples_outputs/issue1937-3-bmv2-first.p4

```diff
@@ -0,0 +1,58 @@
+#include <core.p4>
+#include <v1model.p4>
+
+header h1_t {
+    bit<8> f1;
+    bit<8> f2;
+}
+
+struct headers_t {
+    h1_t h1;
+}
+
+struct metadata_t {
+}
+
+control foo(out bit<8> x, in bit<8> y=5) {
+    apply {
+        x = y >> 2;
+    }
+}
+
+control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
+    @name("foo") foo() foo_inst;
+    @name("foo") foo() foo_inst_0;
```

Comment:
Similar question here and for the parser case: Is it good/important/useful that the compiler is creating two instantiations of control foo?  If foo contained stateful elements like tables, counters, meters, etc., this would be instantiating two of each of those things, yes?

Reply:
Yes, the meaning of invoking a parser is that you create a new instance which you then invoke. So this is exactly what should happen. If you want the other way you should instantiate the parser once and invoke it twice.

Reply:
Yes, sorry for my unnecessary comment.  I realized this was a case of a parser or control being auto-instantiated via an apply call, so two instantiations is correct.

The @name annotations on the two instantiations being identical still looks wrong, though.